### ISSUE TITLE
Dashboard visualization options and click behavior are conflicted each other

### ISSUE THREAD

**Issue by duc-ctm-jacobins-digital** [metabase/metabase#14597]
**Describe the bug**
This bug happens when I config a dashboard
- When I set click behavior for a table, the format of the number will be lost, as well as my column rename
- When I reset visualization options to default, the click behavior will be lost

**To Reproduce**
1. Ask a custom question => Sample dataset => table `order` => Sum of Subtotal by Product → Category
![image](https://user-images.githubusercontent.com/70874681/106412705-ec68a200-647a-11eb-9507-66dbea650be9.png)

2. Format the question: 
- Rename column `Product → Category` to `Product Category`
- Rename column `Sum of Subtotal` to `Subtotal`
- `Sum of Subtotal` to 0 decimal, USD currency
![image](https://user-images.githubusercontent.com/70874681/106412873-479a9480-647b-11eb-8096-a94ff9d4b7f2.png)

3. Create a dashboard and add the question above. Also create a filter named `Product Category` for product category.

4. Set column `Product Category` to update dashboard filter when click on it

5. Click `Done`, and we will lose the number and column name format
![image](https://user-images.githubusercontent.com/70874681/106413296-503f9a80-647c-11eb-8f2c-8dd946aa9782.png)

6. Set visualization options of the table to default, and we will lose the click behavior settings.

**Expected behavior**
When updating click behavior of a table, the visualization options must be remaining.
When resetting visualization options to default, the click behavior must be remaining

**Screenshots**
This gif shows better the steps 5 and 6.
![Test · Dashboard · Metabase (1)](https://user-images.githubusercontent.com/70874681/106414351-fdb3ad80-647e-11eb-8a0f-9197ac67f91f.gif)

**Information about your Metabase Installation:**

```
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.104 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "Cp1252",
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "1.8.0_231-b11",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_231",
    "java.vm.name": "Java HotSpot(TM) 64-Bit Server VM",
    "java.vm.version": "25.231-b11",
    "os.name": "Windows Server 2012 R2",
    "os.version": "6.3",
    "user.language": "en",
    "user.timezone": "Asia/Bangkok"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "sqlserver",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.1"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.37.8",
      "date": "2021-01-26",
      "branch": "release-x.37.x",
      "hash": "490cea7"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
This bug is annoying and wasting a lot of time for me to update the click behavior, then manually update the chart visualization in the dashboard until it matches the question format.

**Comment by flamber** [comment]
Related to #11829 and #13595

**Comment by paulrosenzweig** [comment]
Closed by #14659

**Comment by duc-ctm-jacobins-digital** [comment]
Hi @paulrosenzweig. Can I know when we fix this issue on master branch? I don't find this fix on 0.38.0.1

**Comment by flamber** [comment]
Hi @duc-ctm-jacobins-digital 
The formatting part (step 5) was fixed on 0.38.0, but I think step 6 might have been overlooked. I have created #14919
